src/tx-format/policy.md

```diff
@@ -0,0 +1,65 @@
+# Policy
+
+```c++
+// index using powers of 2 for efficient bitmasking
+enum PolicyType : uint32 {
+    GasPrice = 1,
+    GasLimit = 2,
+    WitnessLimit = 4,
+    Maturity = 8,
+    MaxFee = 16,
+}
+```
+
+| name   | type                                                                                                         | description     |
+|--------|--------------------------------------------------------------------------------------------------------------|-----------------|
+| `data` | One of [GasPrice](#gasprice), [GasLimit](#gaslimit), [WitnessLimit](#witnesslimit), or [Maturity](#maturity) | Policy data.    |
+
+## GasPrice
+
+| name       | type     | description               |
+|------------|----------|---------------------------|
+| `gasPrice` | `uint64` | Gas price for transaction |
+
+## GasLimit
+
+| name       | type     | description                                                |
+|------------|----------|------------------------------------------------------------|
+| `gasLimit` | `uint64` | Gas limit for transaction (applicable to script execution) |
+
+Transaction is invalid if:
+
+- `gasLimit > MAX_GAS_PER_TX`
+
+## WitnessLimit
+
+| name           | type     | description                                                    |
+|----------------|----------|----------------------------------------------------------------|
+| `witnessLimit` | `uint64` | The maximum amount of witness data allowed for the transaction |
+
+Given helper `len()` that returns the number of bytes of a field.
+
+Transaction is invalid if:
+
+- `len(tx.witnesses) > witnessLimit`
+
+## Maturity
+
+| name       | type     | description                              |
+|------------|----------|------------------------------------------|
+| `maturity` | `uint32` | Block until which tx cannot be included. |
+
+Transaction is invalid if:
+
+- `blockheight() < maturity`
+
+## MaxFee
+
+| name      | type     | description                                  |
+|-----------|----------|----------------------------------------------|
+| `max_fee` | `uint64` | The maximum fee payable by this transaction. |
```

Comment:
Is that value in the base token, or is that value in gas? 

- Base token is good because we can later include storage price there too, but there can be a problem with alternative tokens that allow paying for gas(if we can have more than one token within the transaction to do that).

- In gas is good because it is agnostic to the token, but we will face the problem when we introduce a separate storage limit and storage price.

So it seems the first option with value in the base token and rule that we can't use different assets to pay for the transaction.